README.md

```diff
@@ -190,10 +190,27 @@ If you don’t need unpredictable IDs, but you need Web Workers support,
 you can use non‑secure ID generator.
 
 ```js
-const nanoid = require('nanoid/non-secure')
+const nanoid = require('nanoid/dist/cjs/non-secure')
 model.id = nanoid() //=> "Uakgb_J5m9g-0JDMbcJqLJ"
 ```
 
+### ES Modules
+
+in the browser, nanoid can now be loaded via es modules
+
+es modules are available under `nanoid/dist/`
+
+commonjs modules are available under `nanoid/dist/cjs/`
+
+[codepen example](https://codepen.io/ChaseMoskal/pen/XGQBBy?editors=1000)
+
+```html
+<script type="module">
+  import nanoid from "https://unpkg.com/nanoid-esm@2.1.0-esm.1/dist/index.browser.js"
```

Comment:
Using `unpkg.com` is a bad practice since it will add DNS request, TCP connection and one more internet latency to the website loading process. It will slow down the website up to 0.5s.

1. Browser loads `example.com`
2. DNS request, TCP connection, Internet latency
3. Browser see `example.com/index.js` link and make new request.
4. Internet latency
5. Browser see `https://unpkg.com/nanoid-esm@2.1.0-esm.1/dist/index.browser.js` link
6. **DNS request, TCP connection, Internet latency**

With npm and Parcel/webpack you will not have this step 6.

Reply:
an excellent point of course

i think tools like unpkg and jsdelivr are best used during development

optimization options include
- using rollup or similar to create an optimized tree-shaken bundle
- using a lot of [`<link rel="modulepreload" href="..."/>`](https://www.chromestatus.com/feature/5762805915451392) to give the browser a heads-up